libsolidity/analysis/TypeChecker.cpp

```diff
@@ -3358,6 +3358,29 @@ bool TypeChecker::visit(MemberAccess const& _memberAccess)
 
 			annotation.isPure = isPure;
 		}
+	if (
+		auto const* functionType = dynamic_cast<FunctionType const*>(exprType);
+		!annotation.isPure.set() &&
+		functionType &&
+		functionType->kind() == FunctionType::Kind::Event &&
+		functionType->hasDeclaration() &&
+		memberName == "selector"
+	)
+		if (
+			auto const* eventDefinition = dynamic_cast<EventDefinition const*>(&functionType->declaration());
+			eventDefinition &&
+			!eventDefinition->isAnonymous()
+		)
```

Comment:
I don't think this check is needed. An anonymous event does not have a selector member so I think this function will stop with a fatal type error before we even get here.

Also, I see you don't actually have a syntax test case covering this. Please add one.